File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        left = 0
        right = len(heights) - 1
        max_val = 0

        while left != right:
            length = right - left
            if heights[left] > heights[right]:
                temp = length * heights[right]
                right -= 1
            else:
                temp = length * heights[left]
                left += 1
            if max_val > temp:
                continue
            else:
                max_val = temp
        return max_val
            

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    int kthSmallest(TreeNode* root, int k) {
        vector<int> temp(2);
        temp[0] = k;
        dfs(root, temp);
        return temp[1];
    }

    void dfs(TreeNode* r, vector<int>& p){
        if (!r){
            return;
        }
        dfs(r->left, p);
        if (p[0] == 0){return;}

        p[0]--;
        if (p[0] == 0){
            p[1] = r->val;
            return;
        }
        dfs(r->right, p);

    }
};
